### PR TITLE
Adds custom annoucment with custom sound verb

### PR DESCRIPTION
**Pull request by Karolis2011**
This is reused code from huge solar flare event.

### CODE REVIEW

**Comment by Karolis2011** [comment]
!review

File: code/modules/admin/admin.dm [hunk 1]
```diff
@@ -1353,3 +1353,38 @@ proc/admin_notice(var/message, var/rights)
 	log_admin("[key_name(usr)] toggled the round spookyness to [enabled_spooking].")
 	message_admins("[key_name_admin(usr)] toggled the round spookyness [enabled_spooking ? "on" : "off"].", 1)
 	feedback_add_details("admin_verb","SPOOKY") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+
+/datum/admins/proc/ccannoucment()
+	set category = "Special Verbs"
+	set name = "Custom sound Command Announcment"
+	set desc="Emulate announcment that look and sounds like real one."
+	if(!check_rights(R_ADMIN))	return
```
Comment (Arrow768) [inline code/modules/admin/admin.dm]: You should tie it to R_FUN since that's a bit more applicable

File: code/modules/admin/admin.dm [hunk 2]
```diff
@@ -1353,3 +1353,38 @@ proc/admin_notice(var/message, var/rights)
...
+	set name = "Custom sound Command Announcment"
+	set desc="Emulate announcment that look and sounds like real one."
+	if(!check_rights(R_ADMIN))	return
+
+	var/message = input("Announcment content:", "CAnnoucnce", null, null) as message
+	if(!message)
+		return
+	if(!check_rights(R_SERVER,0))
+		message = sanitize(message, 500, extra = 0)
+	var/title = input("Announcment TITLE:", "CAnnoucnce", null, null) as text
```
Comment (Arrow768) [inline code/modules/admin/admin.dm]: Title should be asked before content to be in line with the other verbs

File: code/modules/admin/admin.dm [hunk 3]
```diff
@@ -1353,3 +1353,38 @@ proc/admin_notice(var/message, var/rights)
...
 	message_admins("[key_name_admin(usr)] toggled the round spookyness [enabled_spooking ? "on" : "off"].", 1)
 	feedback_add_details("admin_verb","SPOOKY") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+
+/datum/admins/proc/ccannoucment()
+	set category = "Special Verbs"
+	set name = "Custom sound Command Announcment"
+	set desc="Emulate announcment that look and sounds like real one."
+	if(!check_rights(R_ADMIN))	return
+
+	var/message = input("Announcment content:", "CAnnoucnce", null, null) as message
```
[the remark this answers is not in the document]
Comment (NonQueueingMatt) [reply code/modules/admin/admin.dm]: also "CAnnoucne"

File: code/modules/admin/admin_verbs.dm [hunk 4]
```diff
@@ -98,7 +98,8 @@ var/list/admin_verbs_admin = list(
 	/client/proc/clear_toxins,
 	/client/proc/wipe_ai,	// allow admins to force-wipe AIs
 	/client/proc/fix_player_list,
-	/client/proc/reset_openturf
+	/client/proc/reset_openturf,
+	/datum/admins/proc/ccannoucment
```
Comment (Arrow768) [inline code/modules/admin/admin_verbs.dm]: Shouldnt that be in the `admin_verbs_fun`list?

File: code/modules/admin/admin.dm [hunk 5]
```diff
@@ -1353,3 +1353,39 @@ proc/admin_notice(var/message, var/rights)
...
+	sounds += "--CANCEL--"
+	sounds += "--LOCAL--"
+	sounds += sounds_cache
+
+	var/melody = input("Select a sound from the server to play", "Server sound list", "--CANCEL--") in sounds
+
+	if(melody == "--CANCEL--")
+		return
+	if(melody == "--LOCAL--")
+		melody = input("Select a sound to play", "Sound select") as sound
```
Comment (skull132) [inline code/modules/admin/admin.dm]: Is this missing an `in x` marker? Not sure how well we can expect admins to know this shit by heart.

File: code/modules/admin/admin.dm [hunk 6]
```diff
@@ -1353,3 +1353,39 @@ proc/admin_notice(var/message, var/rights)
 	log_admin("[key_name(usr)] toggled the round spookyness to [enabled_spooking].")
 	message_admins("[key_name_admin(usr)] toggled the round spookyness [enabled_spooking ? "on" : "off"].", 1)
 	feedback_add_details("admin_verb","SPOOKY") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+
+/datum/admins/proc/ccannoucment()
+	set category = "Special Verbs"
+	set name = "Custom sound Command Announcment"
+	set desc = "Emulate announcement that looks and sounds like the real one"
+	if(!check_rights(R_FUN))	return
```
Comment (skull132) [inline code/modules/admin/admin.dm]:
```suggestion
	if(!check_rights(R_FUN))
		return
```

File: code/modules/admin/admin.dm (hunk 5 above)
Comment (Karolis2011) [reply code/modules/admin/admin.dm]: This isn't missing `in x` is just shows file open dialog that allows to select sounds. As of local, it means sound is stored locally on admins computer. Same interface is used in play sound verb.

**Pull request merged**